src/test/spec/unified_runner/matcher.rs

```diff
@@ -39,6 +46,74 @@ pub(crate) fn events_match(
     }
 }
 
+#[cfg(feature = "tracing-unstable")]
+pub(crate) fn tracing_events_match(
+    actual: &TracingEvent,
+    expected: &ExpectedMessage,
+) -> Result<(), String> {
+    if actual.target != expected.target {
+        return Err(format!(
+            "Expected and actual tracing event components do not match: expected {}, got {}",
+            expected.target, actual.target
+        ));
+    }
+    if actual.level != expected.level {
+        return Err(format!(
+            "Expected and actual tracing event levels do not match: expected {}. got {}",
+            expected.level, actual.level
+        ));
+    }
+
+    if let Some(failure_is_redacted) = expected.failure_is_redacted {
```

Comment:
copying from a now outdated thread for visibility: this logic has been updated. note that we redact a few different error types now and not just command errors. before I go down the path of writing regexes matching the Display impl for each possible type of error that gets redacted, do we think that is worthwhile or have any other ideas for how to make these assertions a bit more robust?

Reply:
> Alternatively, we could just replace the entire Display implementation of redacted errors with "REDACTED" and then perform a strict equality, though we'd lose some information that way. Given that this redaction is pretty rare (it's just for when the user uses run_command to create users and what not, right?), I don't think it would be the worst thing in the world.

so we decided in the spec that drivers should not redact error code, error code name, and error labels, so I don't think we'd want to take this approach as we'd strip out everything.

what I ended up doing: 
* writing regexes here for the two types of errors that actually show up and have redaction asserted on them in spec tests; using those to extract components of the errors and assert redaction/lack thereof
* adding unit tests in test/spec/trace.rs that construct all the various types of server errors we redact, call `redact()` on them, then assert that those properties have been redacted as expected

Reply:
LGTM